### PR TITLE
Implement Proximal operator for 1D total variation (fussed lasso)

### PR DESCRIPTION
**Pull request by fabianp**
+ Example
  + Tests

Output of the example:

image

### CODE REVIEW

File: examples/plot_1d_total_variation.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+"""
+Signal recovery using 1D total variation (also known as fussed lasso)
```
Comment (mblondel) [inline examples/plot_1d_total_variation.py]: fused

File: examples/plot_1d_total_variation.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+In this example, we generate a signal that is piecewise constant. We then
+observe some random and corrputed measurements from that signal and
```
Comment (mblondel) [inline examples/plot_1d_total_variation.py]: corrupted

File: lightning/impl/fista.py [hunk 3]
```diff
@@ -146,6 +148,7 @@ class FistaClassifier(BaseClassifier, _BaseFista):
         - l2: ridge
         - l1: lasso
         - l1/l2: group lasso
+        - tv1d: 1-dimensional total variation (also known as fussed lasso)
```
Comment (mblondel) [inline lightning/impl/fista.py]: fused, same below

**Comment by mblondel** [comment]
great work +1 for merge

**Comment by fabianp** [comment]
Great, thanks for the review @mblondel

**Pull request merged**